### DISCUSSION TITLE
How safe are the keys/values? Can someone extract them?

### DISCUSSION THREAD

**Discussion by renatop7** [goatandsheep/react-native-dotenv#509] [Q&A] [answered]
My question is more related to android I guess...
Can someone reverse engineer the .apk and extract my .env variables values?

**Comment by nik27** [comment, marked as answer]
Yes. You should store API keys and other sensitive info on your backend and fetch it after user is authenticated. 

In the readme of [react-native-config](https://github.com/lugg/react-native-config) it's explicitly mentioned:

> Keep in mind this module doesn't obfuscate or encrypt secrets for packaging, so do not store sensitive keys in .env. It's [basically impossible to prevent users from reverse engineering mobile app secrets](https://rammic.github.io/2015/07/28/hiding-secrets-in-android-apps/), so design your app (and APIs) with that in mind.

This is also mentioned in [React Native docs](https://reactnative.dev/docs/security).
[This article](https://www.bam.tech/article/managing-environment-variables-securely-in-react-native) might also be helpful.

**Reply by renatop7** [reply]
Thanks for the inputs :)
